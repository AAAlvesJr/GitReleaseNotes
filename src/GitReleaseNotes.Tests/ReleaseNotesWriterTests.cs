using System;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using GitReleaseNotes.IssueTrackers.GitHub;
using NSubstitute;
using Xunit;

namespace GitReleaseNotes.Tests
{
    [UseReporter(typeof(DiffReporter))]
    public class ReleaseNotesWriterTests
    {
        private readonly IFileSystem _fileSystem;

        public ReleaseNotesWriterTests()
        {
            _fileSystem = Substitute.For<IFileSystem>();
        }

        [Fact]
        public void ApproveSimpleTests()
        {
            var releaseNotesWriter = new ReleaseNotesWriter(_fileSystem);

            var arguments = new GitReleaseNotesArguments();
            var releaseNotes = new SemanticReleaseNotes(new[]
            {
                new ReleaseNoteItem("Issue 1", "#1", new Uri("http://github.com/org/repo/issues/1"), new string[0]), 
            });

            releaseNotesWriter.WriteReleaseNotes(arguments, releaseNotes);

            var content = _fileSystem.ReceivedCalls().Single(c => c.GetMethodInfo().Name == "WriteAllText").GetArguments()[1];
            Approvals.Verify(content);
        }
    }
}