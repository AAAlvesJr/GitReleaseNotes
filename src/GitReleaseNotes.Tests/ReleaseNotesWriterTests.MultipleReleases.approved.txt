# vNext

 - Issue 1 [#1](http://github.com/org/repo/issues/1) +feature


# 1.2.0 (06 December 2013)

 - Issue 2 [#2](http://github.com/org/repo/issues/2) +feature
 - Issue 3 [#3](http://github.com/org/repo/issues/3) +fix
